lib/taskana-core/src/main/java/pro/taskana/TaskanaEngineConfiguration.java

```diff
@@ -35,221 +35,110 @@
 /**
  * This central class creates the TaskanaEngine and holds all the information about DB and Security.
  * <br>
- * Security is enabled by default.
+ * Security is enabled by default. <br>
+ * All members are immutable, also Lists and Maps and Sets.
  */
-@SuppressWarnings("checkstyle:OverloadMethodsDeclarationOrder")
 public class TaskanaEngineConfiguration {
```

Comment:
Just a random thought. Rename this class to `TaskanaConfiguration`? 
I think that name is more suitable